harness/tests/cli/util.py

```diff
@@ -151,3 +153,14 @@ def expect_get_info(
         rsps.get(f"{master_url}/info", status=200, json={"version": det.__version__})
     else:
         responses.get(f"{master_url}/info", status=200, json={"version": det.__version__})
+
+
+def check_cli_output(args: List[str], expected: str) -> None:
+    """
+    Helper method to test CLI methods that checks redirected STDOUT from the executed command
+    matches expected output.
+    """
+    with contextlib.redirect_stdout(io.StringIO()) as f:
+        cli.main(args=args)
+    actual = f.getvalue()
+    assert expected == actual, "CLI output does not match expected output."
```

Comment:
If you are going to make a helper function out of this, you need better error logs.

Try this:
```python
import difflib

if expected == actual:
  return
exp_lines = expected.splitlines(keepends=True)
act_lines = actual.splitlines(keepends=True)
diff_lines = difflib.ndiff(act_lines, exp_lines)
diff = ''.join(diff_lines)
raise ValueError(f"cli output for {args} output actual (-) != expected (+):\n{diff}")

Reply:
Yeah, because of course python stdlib has a diffing tool.

Don't tell anybody, but there's a complete implementation of determined in the python stdlib too.  It's from the 90's though so nobody remembers to look there.